OrbitGl/GpuTrack.cpp

```diff
@@ -103,15 +137,28 @@ float GpuTrack::GetYFromTimer(const TimerInfo& timer_info) const {
   if (collapse_toggle_->IsCollapsed()) {
     adjusted_depth = 0.f;
   }
-  return pos_[1] - layout.GetTextBoxHeight() * (adjusted_depth + 1.f) -
-         adjusted_depth * layout.GetSpaceBetweenGpuDepths();
+  if (timer_info.type() == TimerInfo::kGpuDebugMarker) {
+    return pos_[1] - layout.GetTextBoxHeight() * (adjusted_depth + 1.f);
+  }
+  CHECK(timer_info.type() == TimerInfo::kGpuActivity ||
+        timer_info.type() == TimerInfo::kGpuCommandBuffer);
+
+  if (has_vulkan_layer_command_buffer_timers_) {
+    adjusted_depth *= 2.f;
+  }
+
+  float gap_space = adjusted_depth * layout.GetSpaceBetweenGpuDepths();
+  if (timer_info.type() == TimerInfo::kGpuCommandBuffer) {
+    ++adjusted_depth;
```

Comment:
Nit: To my eye, this looks a bit funky, since this is a floating point value. I would write "adjusted_depth += 1.f". (There's no difference, so this is just a preference.)

Reply:
Maybe more actionable: It's not clear to me why this +1 is needed here. Can you add a comment to explain this a bit? Is this also correct in the collapsed state?

Reply:
Done. Regarding the collapsed state: Yes that is handled correctly. We are not drawing command buffer timers, so we won't ask for the depth here. Explicitly tested this case, thanks for raising that question.